fabric-api-base/src/main/java/net/fabricmc/fabric/api/util/TriState.java

```diff
@@ -16,16 +16,197 @@
+	 */
+	public static TriState of(boolean bool) {
+		return bool ? TRUE : FALSE;
+	}
+
+	/**
+	 * Gets a tri-state from a nullable boxed boolean.
+	 *
+	 * @param bool the boolean value
+	 * @return {@link TriState#DEFAULT} if {@code null}.
+	 * Otherwise {@link TriState#TRUE} or {@link TriState#FALSE} depending on the value of the boolean.
+	 */
+	public static TriState of(/* @Nullable */ Boolean bool) {
+		if (bool == null) {
+			return DEFAULT;
+		}
+
+		return TriState.of(bool.booleanValue());
 	}
 
+	/**
+	 * Gets the value of the tri-state.
+	 *
+	 * @return true if the tri-state is {@link TriState#TRUE}.
+	 * Otherwise false.
+	 */
 	public boolean get() {
 		return this == TRUE;
 	}
+
+	/**
+	 * Gets the value of the tri-state as a boxed, nullable boolean.
+	 *
+	 * @return {@code null} if {@link TriState#DEFAULT}.
+	 * Otherwise {@code true} if {@link TriState#TRUE} or {@code false} if {@link TriState#FALSE}.
+	 */
+	/* @Nullable */
+	public Boolean getBoxed() {
+		if (this == TriState.DEFAULT) {
+			return null;
+		}
+
+		return this.get();
+	}
+
+	/**
+	 * Gets the value of this tri-state.
+	 * If the value is {@link TriState#DEFAULT} then use the supplied value.
+	 *
+	 * @param value the value to fallback to
+	 * @return the value of the tri-state or the supplied value if {@link TriState#DEFAULT}.
+	 */
+	public boolean orElse(boolean value) {
+		return this == TriState.DEFAULT ? value : this.get();
+	}
+
+	/**
+	 * Gets the value of this tri-state.
+	 * If the value is {@link TriState#DEFAULT} then use the supplied value.
+	 *
+	 * @param supplier the supplier used to get the value to fallback to
+	 * @return the value of the tri-state or the value of the supplier if the tri-state is {@link TriState#DEFAULT}.
+	 */
+	public boolean orElseGet(BooleanSupplier supplier) {
+		return this == TriState.DEFAULT ? supplier.getAsBoolean() : this.get();
+	}
+
+	/**
+	 * Maps the value of an optional if this tri-state is {@link TriState#TRUE}.
+	 *
+	 * @param mapper the mapper to use
+	 * @param <T> the type of object supplied by the mapper
+	 * @return an optional containing the value if {@link TriState#TRUE} and the value returned by the mapper is non-null, otherwise {@link Optional#empty()}.
+	 */
+	public <T> Optional<T> mapTrue(Supplier<T> mapper) {
+		Objects.requireNonNull(mapper);
+
+		if (this == TRUE) {
+			return Optional.ofNullable(mapper.get());
+		}
+
+		return Optional.empty();
+	}
+
+	/**
+	 * Maps the value of an optional if this tri-state is {@link TriState#TRUE}.
+	 *
+	 * @param mapper the mapper to use
+	 * @param <T> the type of object supplied by the mapper
+	 * @return an optional containing the value if {@link TriState#FALSE} and the value returned by the mapper is non-null, otherwise {@link Optional#empty()}.
+	 */
+	public <T> Optional<T> mapFalse(Supplier<T> mapper) {
+		Objects.requireNonNull(mapper);
+
+		if (this == FALSE) {
```

Comment:
Sometimes you use TriState.FALSE and other times you just use FALSE. I personally prefer TriState.FALSE just to be explicit, but I would just choose one and stick to that as a general style rule.